### PR TITLE
Apply user's browser date formatting to timestamps [SCI-8778]

### PR DESCRIPTION
**Pull request by sboursen-scinote**
Jira ticket: [SCI-8778](https://scinote.atlassian.net/browse/SCI-8778)

### What was done
Extract Date formatting from the user browser in shareable links views and apply it to shareable links View dates.

[SCI-8778]: https://scinote.atlassian.net/browse/SCI-8778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/views/shareable_links/my_modules/_step.html.erb [hunk 1]
```diff
@@ -43,7 +43,7 @@
   <div class="collapse in" id="stepBody<%= step.id %>">
     <div class="step-elements">
       <div class="step-timestamp">
-        <%= t('protocols.steps.timestamp', date: step.created_at, user: step.user.full_name) %>
+        <%= t('protocols.steps.timestamp_iso', date: l(step.created_at, format: :full_iso), user: step.user.full_name).html_safe %>
```
Comment (okriuchykhin) [inline app/views/shareable_links/my_modules/_step.html.erb:46]: don't use html_safe

File: app/assets/javascripts/shareable_links/date_formatting.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+(function() {
+  $('.iso-formatted-date').each(function() {
+    let text = $(this).text().trim();
+
+    if (!text) {
+      return;
+    }
+
+    // Convert to ISO format for js
+    text = `${text.slice(0, -2)}:${text.slice(-2)}`;
```
Comment (okriuchykhin) [inline app/assets/javascripts/shareable_links/date_formatting.js:12]: why do you need this?

File: app/views/shareable_links/my_modules/_start_date_label.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<span class="start-date-label">
+  <% if my_module.started_on.present? %>
+    <span class="iso-formatted-date font-bold"><%= l(my_module.started_on, format: :full_iso) %></span>
```
Comment (okriuchykhin) [inline app/views/shareable_links/my_modules/_start_date_label.html.erb:3]: I think you can simply use my_module.started_on.iso8601

File: app/views/shareable_links/my_modules/_due_date_label.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<span class="due-date-label <%= get_task_alert_color(my_module) %>">
+  <% if (my_module.archived_branch? || my_module.completed? ) && my_module.due_date %>
+    <span class="iso-formatted-date"><%= l(my_module.due_date, format: :full_iso) %></span>
+  <% elsif my_module.is_one_day_prior? %>
+    <%= t('my_modules.states.due_soon') %>
+    <span class="iso-formatted-date"><%= l(my_module.due_date, format: :full) %></span>
```
Comment (okriuchykhin) [inline app/views/shareable_links/my_modules/_due_date_label.html.erb:6]: not in iso format?

File: app/views/shareable_links/my_modules/_start_date_label.html.erb (hunk 3 above)
Comment (sboursen-scinote) [reply app/views/shareable_links/my_modules/_start_date_label.html.erb:3]: Yes that's right, but we will have to handle it differently in JS

File: app/helpers/shareable_link_helper.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module ShareableLinkHelper
+  def iso_datetime_picker_format_date_only
+    'YYYY-MM-DD'
+  end
+
+  def iso_datetime_picker_format_full
```
Comment (okriuchykhin) [inline app/helpers/shareable_link_helper.rb:8]: is it used anywhere?

**Review by okriuchykhin — APPROVED** [review]

**Pull request merged**